web/packages/teleport/src/UnifiedResources/ResourceActionButton.tsx

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { ButtonBorder } from 'design';
+import { LoginItem, MenuLogin } from 'shared/components/MenuLogin';
+import { DbProtocol } from 'shared/services/databases';
+
+import { UnifiedResource } from 'teleport/services/agents';
+
+import AwsLaunchButton from 'teleport/Apps/AppList/AwsLaunchButton';
+import { Database } from 'teleport/services/databases';
+import { App } from 'teleport/services/apps';
+import { Kube } from 'teleport/services/kube';
+import { Desktop } from 'teleport/services/desktops';
+
+type Props = {
+  resource: UnifiedResource;
+  getNodeLoginOptions: (serverId: string) => LoginItem[];
+  getWindowsLoginOptions: (desktop: Desktop) => LoginItem[];
+  startSshSession: (login: string, serverId: string) => void;
+  startRemoteDesktopSession: (username: string, desktopName: string) => void;
+  setDbConnectInfo: React.Dispatch<
+    React.SetStateAction<{
```

Comment:
Nit: the fact that we are linking to a `useState` setter here is an implementation detail of the component that owns a `ResourceActionButton`. So even if technically it's possible to use a reducer syntax here and technically, `React.Dispatch` is a declared type of a `useState` hook result, specifying it on this level is unnecessary. The declared interface should be as thin as possible, so I'd argue that simply using a `connectToDb: (name, protocol) => void` will be much cleaner.